### DISCUSSION TITLE
Google Translate Widget

### DISCUSSION THREAD

**Discussion by BhathiyaPrasad** [community/community#136638] [Programming Help] [answered]
### Google Translate widget for nextjs with TypeScript
1. Is it possible to add google translate button to nextjs app
  I tried So many Times, but it did not work it works fine with plain html 
### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ferid333** [comment]
Yes, you can add a Google Translate button to a Next.js app using TypeScript, but it's not as straightforward as with plain HTML. Here's a simple way to do it:

1. **Add Google Translate Script**: First, include the Google Translate script in your `_document.tsx` file. This is where you can add custom scripts for your entire app.

    ```typescript
    import Document, { Html, Head, Main, NextScript } from 'next/document';

    class MyDocument extends Document {
      render() {
        return (
          <Html>
            <Head>
              {/* Add the Google Translate script */}
              <script
                dangerouslySetInnerHTML={{
                  __html: `
                    function googleTranslateElementInit() {
                      new google.translate.TranslateElement(
                        {pageLanguage: 'en'},
                        'google_translate_element'
                      );
                    }
                  `,
                }}
              />
              <script
                src="https://translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"
                async
              />
            </Head>
            <body>
              <Main />
              <NextScript />
              <div id="google_translate_element"></div>
            </body>
          </Html>
        );
      }
    }

    export default MyDocument;
    ```

2. **Add a Translate Button**: Place a button or a dropdown in your app where you want the user to trigger the translation.

    ```typescript
    import React from 'react';

    const TranslateButton: React.FC = () => {
      return (
        <button onClick={() => {
          // Trigger the translation
          (window as any).googleTranslateElementInit();
        }}>
          Translate
        </button>
      );
    };

    export default TranslateButton;
    ```

3. **Use the Button in Your Pages**: You can now use this button in your Next.js pages or components.

    ```typescript
    import TranslateButton from '../components/TranslateButton';

    const HomePage: React.FC = () => {
      return (
        <div>
          <h1>Welcome to My App</h1>
          <TranslateButton />
        </div>
      );
    };

    export default HomePage;
    ```

This method loads the Google Translate widget on your page. Make sure to test it thoroughly, as Google Translate might have some limitations or require additional configurations depending on your specific needs.

**Comment by BhathiyaPrasad** [comment]
I will try it thankyou

**Reply by ferid333** [reply]
You are welcome. Please close the discussion after that Thanks

**Comment by BhathiyaPrasad** [comment, marked as answer]
I tried and make a library component it available in my repository list
